spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java

```diff
@@ -59,7 +60,8 @@ public Azure azure() throws IOException {
     @Bean
     @ConditionalOnProperty(name = "spring.cloud.azure.telemetryAllowed", havingValue = "true", matchIfMissing = true)
     public TelemetryTracker telemetryTracker(Azure azure, AzureProperties azureProperties,
-                                             TelemetryProperties telemetryProperties) {
-        return new TelemetryTracker(azure, azureProperties.getResourceGroup(), telemetryProperties);
+            TelemetryProperties telemetryProperties) {
+        return new TelemetryTracker(azure.getCurrentSubscription().subscriptionId(), azureProperties.getResourceGroup(),
+                    telemetryProperties.getInstrumentationKey());
```

Comment:
passing properties instead of `key`, if there may be another property from telemetry, then you cannot add new args.

Reply:
As I see for now, key is only property required for initializing this tracker. So I prefer not rely on properties.